src/IonImport.ts

```diff
@@ -28,39 +28,49 @@ import { SymbolIndex } from "./IonSymbolIndex";
 export class Import {
     private readonly _offset: number;
     private readonly _length: number;
-    private readonly index: SymbolIndex = {};
     private readonly _parent : Import;
     private readonly _symbolTable : SharedSymbolTable;
 
     constructor(parent : Import, symbolTable : SharedSymbolTable, length?: number) {
         this._parent = parent;
         this._symbolTable = symbolTable;
         this._offset = this.parent ? this.parent.offset + this.parent.length : 1;
-        this._length = length || this.symbolTable.symbols.length;
-
-        let symbols: string[] = this.symbolTable.symbols;
-        for (let i: number = 0; i < this.length; i++) {
-            this.index[symbols[i]] = this.offset + i;
-        }
+        this._length = length || this.symbolTable.numberOfSymbols;
     }
 
-    getSymbol(symbolId: number) : string {
+    getSymbolText(symbolId: number) : string {
         if(this.parent === undefined) throw new Error("Illegal parent state.");
         if (this.parent !== null) {
-            let parentSymbol = this.parent.getSymbol(symbolId);
+            let parentSymbol = this.parent.getSymbolText(symbolId);
             if (parentSymbol) {
                 return parentSymbol;
             }
         }
 
         let index: number = symbolId - this.offset;
-        if (index < this.length) {
-            return this.symbolTable.symbols[index];
+        if (index >= 0 && index < this.length) {
+            return this.symbolTable.getSymbolText(index);
         }
+        return undefined;
```

Comment:
Do we really want this to return `undefined`, i.e. do we really want this to be the contract?

Reply:
anywhere we can ripout undefined we should do so

Reply:
As noted in the `Out of Scope` section of the overview, we should replace all instances of `undefined` with an `UNKNOWN_SYMBOL` constant. Ideally, this would be combined with the introduction of a proper `SymbolToken` class.